src/gui_common/ChemicalEquation.cs

```diff
@@ -77,6 +83,17 @@ public bool ShowSpinner
         }
     }
 
+    public bool ProcessEnabled
+    {
+        get => toggleProcess.ButtonPressed;
+        set
+        {
+            EmitSignal(SignalName.ToggleProcessPressed, this);
```

Comment:
This should check if the new value is the same as the old one and not trigger the signal if that is the case.